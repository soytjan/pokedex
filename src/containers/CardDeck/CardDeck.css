.CardDeck {
  background-color: floralwhite;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: auto;
  grid-gap: 15px 10px;
  min-width: 1100px;
}