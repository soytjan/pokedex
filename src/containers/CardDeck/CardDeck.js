import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addPokemon, updatePokemon, toggleSelected } from '../../actions';
import { fetchPokemonType, fetchAllPokemon } from '../../helper';
import pikachu from '../../loading.gif';
import Card from '../../components/Card/Card';
import './CardDeck.css';

export class CardDeck extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
    }
  }

  // refactor this to break this out into different functions
  componentDidMount = async () => {
    this.setState({ isLoading: true })
    await this.getInitialPokemonTypes();
    this.setState({ isLoading: false })
    this.getPokemonDetails(this.props.pokemon);
  }

  getInitialPokemonTypes = async () => {
    const { addPokemon } = this.props;
    const pokemonType = await fetchPokemonType();
    addPokemon(pokemonType);
  }

  getPokemonDetails = async (pokemon) => {
    const { updatePokemon } = this.props;
    const allPokemon = await fetchAllPokemon(pokemon);
    updatePokemon(allPokemon);
  } 

  handleCardClick = (card) => {
    const { toggleSelected } = this.props;
    const updatedCard = {...card, isSelected: !card.isSelected}
    toggleSelected(updatedCard);
  }

  renderCards = () => {
    const { pokemon } = this.props;
    return pokemon.map(pokemon => {
      return <Card 
        pokemon={pokemon}
        onClick={this.handleCardClick}
        key={pokemon.id} 
      />
    })
  }

  render() {
    if (this.state.isLoading) {
      return (
        <div className='gif'>
          <img src={pikachu} alt="pikachu loading gif"/>
        </div>
      )
    }

    return (
      <section className='CardDeck'>
        I'm the CardDeck!
        {this.renderCards()}
      </section>
    )
  }
}

CardDeck.propTypes = {
  addPokemon: PropTypes.func,
  updatePokemon: PropTypes.func,
  pokemon: PropTypes.array
};

export const mapStateToProps = state => ({
  pokemon: state.pokemon,
})

export const mapDispatchToProps = dispatch => ({
  addPokemon: (pokemon) => dispatch(addPokemon(pokemon)),
  updatePokemon: (pokemon) => dispatch(updatePokemon(pokemon)),
  toggleSelected: (card) => dispatch(toggleSelected(card))
})

export default connect(mapStateToProps, mapDispatchToProps)(CardDeck);