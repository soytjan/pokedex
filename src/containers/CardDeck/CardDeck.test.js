import React from 'react';
import { shallow } from 'enzyme';
import { CardDeck } from './CardDeck';

describe('CardDeck', () => {
  it('should return false', () => {
    expect(true).toEqual(false);
  })
})