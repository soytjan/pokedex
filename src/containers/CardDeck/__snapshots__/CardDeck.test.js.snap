// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDeck should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardDeck
    addPokemon={[MockFunction]}
    pokemon={
      Array [
        Object {
          "id": "1",
          "name": "normal",
          "pokemon": Array [
            "16",
            "17",
            "18",
            "19",
            "20",
          ],
        },
      ]
    }
    toggleSelected={[MockFunction]}
    updatePokemon={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="pikachu loading gif"
        src="loading.gif"
      />,
      "className": "gif",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "pikachu loading gif",
        "src": "loading.gif",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="pikachu loading gif"
          src="loading.gif"
        />,
        "className": "gif",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "pikachu loading gif",
          "src": "loading.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
